### PR TITLE
JIRA Transition Action

### PR DESCRIPTION
**Pull request by valeriomazzeo**
Implemented custom action to automate JIRA issues transitions based on issue ids found in the changelog.

This is particularly useful when running fastlane on a CI, for example when deploying to HockeyApp / Crashlytics, to move all the deployed issues to another state (for example: Ready For Testing).

### CODE REVIEW

File: fastlane/lib/fastlane/actions/jira_transition.rb [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+        issue_ids = Actions.lane_context[SharedValues::FL_CHANGELOG].scan(/#{project_key}-\d+/i).uniq
+
+        issue_ids.each do |issue_id|
+          begin
+            issue = client.Issue.find(issue_id)
+            transition = issue.transitions.build
+            transition.save!("transition" => { "id" => transition_id })
+
+            if !comment_text.nil?
```
Comment (valeriomazzeo) [inline fastlane/lib/fastlane/actions/jira_transition.rb]: for some reasons it wasn't working, but I am not an expert in ruby...

File: fastlane/lib/fastlane/actions/jira_transition.rb [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+              comment = issue.comments.build
+              comment.save({"body" => comment_text})
+            end
+          rescue Exception => e
```
Comment (valeriomazzeo) [inline fastlane/lib/fastlane/actions/jira_transition.rb]: not sure what to do here, some help would be appreciated

**Comment by KrauseFx** [comment]
Hey @valeriomazzeo, thanks for the pull request. We just launched the fastlane plugin system, that allows you to build your own plugins without having to merge it into the main code base.

Get started here: https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Plugins.md#create-your-own-plugin
You can basically copy over the action you already built after running `fastlane new_plugin` 🚀 

Please let us know how that works for you and if there is any way we can help 👍

**Comment by valeriomazzeo** [comment]
@KrauseFx the main point of this was that it could be useful to other people too, is there anywhere where are you collecting all these plugins?

**Comment by valeriomazzeo** [comment]
Hey @KrauseFx, no problem, already up and running:
https://rubygems.org/gems/fastlane-plugin-jira_transition/versions/0.1.0
https://github.com/valeriomazzeo/fastlane-plugin-jira_transition
🎉

**Comment by KrauseFx** [comment]
This is great news, thanks @valeriomazzeo, how was the onboarding process? How can we improve the docs or make it easier in general? 👍

**Comment by valeriomazzeo** [comment]
I didn't have any trouble and it went pretty smoothly.
The only issue I had it was with writing the action itself, especially when trying to call another action from inside the action.

The problem was with the parameters, and I solved it creating a configuration item, but I haven't seen you doing that anywhere else in the code, so I am not too sure about it.

The repo is https://github.com/valeriomazzeo/fastlane-plugin-jira_transition

**Comment by KrauseFx** [comment]
Great to hear. Here's some docs on how to call actions from other actions: https://github.com/fastlane/fastlane/tree/master/fastlane/docs#extensions

**Pull request closed without merging**